tspp.java

```diff
@@ -54,19 +57,21 @@ void dfs_1()
 {
 Stack <Integer> s=new Stack<Integer> ();
 System.out.println("Enter the starting vertex: ");
-int start=sc.nextInt();
+
+@LTLengthOf("visited") int start=sc.nextInt();
 int visited[]=new int[v];
 
 s.push(start);
 visited[start]=1;
 System.out.print("Path :    "+start+"     ");
-int vertex=1,cost=0;
+@LTLengthOf("visited") int vertex=1;
+int cost=0;
 int min=Integer.MAX_VALUE;
 
 while(!s.isEmpty())
 {
 int flag=0;
-int ele=s.peek();
+@LTLengthOf("mat") int ele=s.peek();
```

Comment:
For this to be true, every element of s should be `@LTLengthOf("mat")`. You can express this using an annotation where you declared the `s` stack.

Reply:
Done!